### PR TITLE
Add helpers to set, clear, and check IsUnusedValue.

### PR DESCRIPTION
**Pull request by pgavlin**
Just what it says on the tin.

### CODE REVIEW

**Comment by pgavlin** [comment]
@CarolEidt @BruceForstall @dotnet/jit-contrib PTAL

File: src/jit/lir.h [hunk 1]
```diff
@@ -309,4 +309,19 @@ class LIR final
...
+inline void GenTree::ClearIsUnusedValue()
+{
+    gtLIRFlags &= ~LIR::Flags::IsUnusedValue;
+}
+
+inline bool GenTree::IsUnusedValue() const
+{
+    return (gtLIRFlags & LIR::Flags::IsUnusedValue) != 0;
+}
+
```
Comment (BruceForstall) [inline src/jit/lir.h]: Naming: I wonder if the flag should be `LIR::Flags::UnusedValue` and the functions thereby `SetUnusedValue` / `ClearUnusedValue` / `IsUnusedValue`? Having the word `Is` on the flag itself seems linguistically wrong, and makes `SetIsUnusedValue` a tongue-twister.

File: src/jit/gentree.h [hunk 2]
```diff
@@ -1169,6 +1169,11 @@ struct GenTree
         }
     }
 
+    // NOTE: these functions are defined in lir.h.
+    inline void SetIsUnusedValue();
```
Comment (BruceForstall) [inline src/jit/gentree.h]: I would be more specific than "these", e.g., "The functions to set/clear/check the UnusedValue bit are defined in lir.h"

**Comment by CarolEidt** [comment]
I would second @BruceForstall's suggestions, but otherwise LGTM.

**Comment by pgavlin** [comment]
The proposed renames sound good; I'll make the change.

**Pull request merged**